iceoryx_dust/source/posix_wrapper/named_pipe.cpp

```diff
@@ -34,124 +35,124 @@ constexpr units::Duration NamedPipe::CYCLE_TIME;
 constexpr units::Duration NamedPipe::NamedPipeData::WAIT_FOR_INIT_SLEEP_TIME;
 constexpr units::Duration NamedPipe::NamedPipeData::WAIT_FOR_INIT_TIMEOUT;
 
-NamedPipe::NamedPipe() noexcept
+expected<NamedPipe, IpcChannelError> NamedPipeBuilder::create() const noexcept
 {
-    this->m_isInitialized = false;
-    this->m_errorValue = IpcChannelError::NOT_INITIALIZED;
-}
-
-// NOLINTNEXTLINE(readability-function-size) @todo iox-#832 make a struct out of arguments
-NamedPipe::NamedPipe(const IpcChannelName_t& name,
-                     const IpcChannelSide channelSide,
-                     const size_t maxMsgSize,
-                     const uint64_t maxMsgNumber) noexcept
-{
-    // We do not store maxMsgSize or maxMsgNumber, this is just technical debt since every ipc channel
-    // requires the same behavior as the message queue. The named pipe would get later two template
-    // parameters MAX_MSG_SIZE and MAX_MSG_NUMBER from which the Message_t size and m_messages queue
-    // size is obtained. Reducing the max message size / number of messages even further would not gain
-    // reduced memory usage or decreased runtime. See issue #832.
-    if (name.size() + strlen(&NAMED_PIPE_PREFIX[0]) > MAX_MESSAGE_SIZE)
+    if (m_name.size() + strlen(&NamedPipe::NAMED_PIPE_PREFIX[0]) > NamedPipe::MAX_MESSAGE_SIZE)
     {
-        std::cerr << "The named pipe name: \"" << name
-                  << "\" is too long. Maxium name length is: " << MAX_MESSAGE_SIZE - strlen(&NAMED_PIPE_PREFIX[0])
-                  << std::endl;
-        m_isInitialized = false;
-        m_errorValue = IpcChannelError::INVALID_CHANNEL_NAME;
-        return;
+        std::cerr << "The named pipe name: \"" << m_name << "\" is too long. Maxium name length is: "
```

Comment:
I think we need an issue to replace all `std::cerr` usages with `IOX_LOG` ... is it somewhere in place?